### PR TITLE
Filter overlay

### LINKED ISSUE #116: Better sanity checking of config defaults
instead of the long list of `if` statements checking for `undefined` in line ~1060 of `toolbox.ts` in 63bd99f, check that config values are of the expected type. could also iterate over a list of defaults instead of separate if statements

### PR DESCRIPTION
**Pull request by csolbs24**
# Add Multi-Class Support to FilterDistanceToolboxItem and Add an Overlay to Show Filter's Range

## Description

Added new features to the `FilterDistanceToolboxItem`. Added the ability to filter with different distances based on the polyline's class. Additionally added an overlay to the ToolboxItem to visualize the distance from the row. Also since many sliders are used across multiple ToolboxItems, I added a new SliderHandler class the allows you to easily create a slider. 

Unsure of some code placement. The largest such example is the new `SliderHandler`. I put it into the `html_builder.ts` file because it assists with the creation of the slider html, but I'm not sure if it should be here, in its own file, or somewhere else entirely. 

Additionally looks to encapsulate some of the toolbox specific css from `blobs.js` into their respective `ToolboxItems`s.

[66e7489](https://github.com/SenteraLLC/ulabel/pull/112/commits/66e748953382db583322ec31a647c6a89b3fc8e8) closes #116 

## PR Checklist

- [x] Merged latest main
- [x] Version number in `package.json` has been bumped since last release
- [x] Version numbers match between package `package.json` and `src/version.js`
- [x] Ran `npm install` and `npm run build` AFTER bumping the version number
- [ ] Updated documentation if necessary (currently just in `api_spec.md`)
- [x] Added changes to `changelog.md`

## Breaking API Changes

Removed `config.filter_row_distance_on_load` and `config.filter_row_distance_default_value` and moved the same functionality into `config.distance_filter_toolbox_item.filter_on_load` and `config.distance_filter_toolbox_item.default_value`

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -18,40 +32,61 @@ export type DeprecatedBy = {
...
-    frame: number;
-    text_payload: string;
-    annotation_meta: any;
-    distance_from_any_line?: number;
-};
+/**
+ * Info needed to filter distance from row without accessing the dom. 
+ * Primarily exists so that points can be filtered before the page loads.
+ */
+export type FilterDistanceOverride = {
```
Comment (joshua-dean) [inline index.d.ts:39]: These look clean, make sure to get them in `api_spec.md`

File: src/annotation_operators.ts [hunk 2]
```diff
@@ -307,15 +311,155 @@ export function filter_points_distance_from_line(ulabel: ULabel, offset: Offset
...
+    if (override === null) {
+        // Used for error checking
+        let return_early: boolean = false
+
+        // Try to grab the elements from the dom
+        const multi_checkbox: HTMLInputElement = document.querySelector("#filter-slider-distance-multi-checkbox")
+        const show_overlay_checkbox: HTMLInputElement = document.querySelector("#filter-slider-distance-toggle-overlay-checkbox")
+        const sliders: NodeListOf<HTMLInputElement> = document.querySelectorAll(".filter-row-distance-slider")
+        
+        // Check to make sure each elemet exists before trying to use
```
Comment (TrevorBurgoyne) [inline src/annotation_operators.ts:351]:
```suggestion
        // Check to make sure each element exists before trying to use
```

File: src/index.js [hunk 3]
```diff
@@ -1077,6 +1091,46 @@ export class ULabel {
             // Draw resumed from annotations
             that.redraw_all_annotations();
 
+            // Update the overlay now that required values have been updated if the overlay exists
```
[the remark this answers is not in the document]
Comment (TrevorBurgoyne) [reply src/index.js:1096]: @csolbs24

File: src/overlays.ts [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+    }
+
+    public createCanvas(canvas_width, canvas_height): void {
+        // Create the canvas element
+        this.canvas = document.createElement("canvas")
+
+        // Add a class to identify created overlays
+        this.canvas.setAttribute("class", "ulabel-overlay")
+
+        // Overlays should on top of everything, so give it a reasonably large z-index
```
Comment (joshua-dean) [inline src/overlays.ts:53]: What if we want overlays to be over the image but under the annotations? 
Is this easily set via the z-index, and if so, is it easily configurable?

**Review by joshua-dean — COMMENTED** [review]
Moving css into each `ToolboxItem` from `blob.js` is a good move.

File: src/overlays.ts (hunk 4 above)
Comment (csolbs24) [reply src/overlays.ts:53]: Yeah its as simple as changing the z-index. We could add an option when creating an overlay to determine whether it should go in front of or behind annotations.

File: api_spec.md [hunk 5]
```diff
@@ -264,6 +266,80 @@ The line width with which new annotations are drawn initially. Units are pixels
 
 URL to a page that gives annotation instructions.
 
+### `config_data`
+
+*object* -- An object to configure much of ULabel's behaviors.
+
+```javascript
```
Comment (joshua-dean) [inline api_spec.md:273]: Did not know this was available for Markdown. Very cool

File: src/overlays.ts (hunk 4 above)
Comment (joshua-dean) [reply src/overlays.ts:53]: Let's do that. If not in this PR, open an issue to have it configurable. 
Definitely would be necessary if using multiple overlays as well.

**Review by joshua-dean — APPROVED** [review]

File: src/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (csolbs24) [reply src/index.js:1013]: The overlay should be initialized inside of the toolbox item now

**Pull request merged**